compiler_opt/rl/env.py

```diff
@@ -260,14 +265,32 @@ def clang_session(
     def _get_scores() -> dict[str, float]:
       return task.get_module_scores(task_working_dir)
 
+    def delete_working_dir(proc, writer_name, reader_name):
```

Comment:
I think this should go in its own PR, maybe it can be the same as the PR for the change to the log_reader (they are quite related, so most probably yes).

You can test by having the fake "clang" the test uses controllably block at each problematic step - you can imagine a few very similar tests, just that each configures the "clang" to fail accordingly.